### ISSUE TITLE
Add a runAsUser option to launch containers with non-root user

### ISSUE THREAD

**Issue by francoisserra** [confluentinc/cp-helm-charts#173]
Some deploying platforms (like OpenShift MustRunAsNonRoot settings) prevent containers to run as root user. There should be a way to specify to run as non-privileged users.

For instance ZK starts well by specifying runAsUser: 104 in the statefulset config.

Should we provide a general option (e.g. runUnprivileged) in the values.yml or offer the possibility to specify the uid to use in each deployment/statefulset of each chart ?

**Comment by gAmUssA** [comment]
@francoisserra would you be kind to send a PR for that?

**Comment by francoisserra** [comment]
I'm struggling with the following commit on the cp-docker images side https://github.com/confluentinc/cp-docker-images/pull/575  : e.g. it has made the folder /var/lib/zookeeper unwritable for non-root users (i.e. cp-kafka user)

**Comment by francoisserra** [comment]
> @francoisserra Those directories are owned by the root group and writable by that group as well. Your user cp-kafka should be part of the root group. We made that change in cp-docker-images specifically to allow the image to be run in OpenShift where their guideline was for directories to be owned/writable by the root group, so any arbitrary user can be used instead of root user. See Support Arbitrary User IDs section in https://docs.openshift.com/container-platform/3.9/creating_images/guidelines.html#openshift-specific-guidelines

You're right, I didn't saw this. By the way I will be able to submit the PR soon. I had to use the supplementalGroups directive to add the root group to the user.

```
$ docker run --rm -ti confluentinc/cp-zookeeper:5.0.0 /usr/bin/id
uid=0(root) gid=0(root) groups=0(root)
$ docker run --rm -ti --user 104 confluentinc/cp-zookeeper:5.0.0 /usr/bin/id
uid=104(cp-kafka) gid=109(confluent) groups=109(confluent)
$ docker run --rm -ti --user 104 --group-add 0 confluentinc/cp-zookeeper:5.0.0 /usr/bin/id
uid=104(cp-kafka) gid=109(confluent) groups=109(confluent),0(root)
```
